luarules/gadgets/unit_area_timed_damage.lua

```diff
@@ -12,484 +12,229 @@ function gadget:GetInfo()
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 200,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_air_bomber_acid_t2_v1_acidbomb'] = {
-        ceg = "acid-area-150",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 100,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_land_swarmer_acids_t2_v1_acidspit'] = {
-        ceg = "acid-area-75",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 40,
-        range = 75,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_allterrain_swarmer_acid_t2_v1_acidspit'] = {
-        ceg = "acid-area-75",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 40,
-        range = 75,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_allterrain_assault_acid_t2_v1_acidspit'] = {
-        ceg = "acid-area-150",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 100,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_turret_acid_t2_v1_acidspit'] = {
-        ceg = "acid-area-150",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 200,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_turret_acid_t3_v1_acidspit'] = {
-        ceg = "acid-area-150",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 200,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_turret_acid_t4_v1_acidspit'] = {
-        ceg = "acid-area-150",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 200,
-        range = 150,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_matriarch_acid_acidgoo'] = {
-        ceg = "acid-area-75",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 40,
-        range = 75,
-        resistance = "_RAPTORACID_",
-    },
-    ['raptor_matriarch_acid_spike_acid_blob'] = {
-        ceg = "acid-area-75",
-        damageCeg = "acid-damage-gen",
-        time = 10,
-        damage = 40,
-        range = 75,
-        resistance = "_RAPTORACID_",
-    },
+local function AddTimedAreaDef(id, def)
+	if not def.customParams or not def.customParams.area_timed_damage == true then return end
+	local isUnitDef = UnitDefNames[def.name] and true or false
 
+	-- SpawnExplosion needs a weapondef entry to do its thing:
+	local areaWeaponDef
+	if isUnitDef then
+		areaWeaponDef = WeaponDefNames[def.name .. "_area_timed_damage"]
+	else
+		areaWeaponDef = WeaponDefNames[def.name .. "_area"]
+		if not areaWeaponDef then
+			local name = ""
+			for word in string.gmatch(def.name, "([^_]+)") do
+				name = name == "" and word or name.."_"..word
+				if UnitDefNames[name] and WeaponDefNames[name.."_area_timed_damage"] then
+					areaWeaponDef = WeaponDefNames[name.."_area_timed_damage"]
+				end
+			end
+		end
```

Comment:
String parsing is fragile and it is easy to accidentally break the code. Use customparams, rather than pattern matching on the name, or even better, use a custom weapon type.